### ISSUE TITLE
Dynamic Linking error

### ISSUE THREAD

**Issue by Damilare06** [SCOREC/wdmapp_coupling#18]
Up until yesterday, I had been working with a previous GENE and XGC install (about 1-month-old) on cori... coupling them via the updated coupler worked perfectly

It appears the recent cleanups in GENE, the CPL adios2 shared librry linking or there must have been a change to the DEFAULT m499 apps - because at the moment camtimers and pspline has to be added to the `LD_LIBRARY_PATH` in order to build XGC. There was no need for this a month ago.

Also, GENE doesn't successfully compile with the DEFAULT adios2, it returns

```/usr/bin/ld: skipping incompatible /lib/libgcc_s.so.1 when searching for libgcc_s.so.1
/usr/bin/ld: bin/obj_cori/adios2_comm_mod.o: in function `adios2_comm_module_mp_adios2_comm_init_':
adios2_comm_mod.F90:(.text+0x1091): undefined reference to `adios2_init_config_'
make[1]: *** [makefiles/rules.mk:343: bin/gene_cori] Error 1
make[1]: Leaving directory '/global/u2/d/damilare/wdmapp/wdmapp_coupling/gene_adios2'
make: *** [makefile:118: bin/gene_cori] Error 2
```

Its compilation succeeds when the adios2 is added to  `LD_LIBRARY_PATH`

Additionally, the `coupler` 's only builds completely with the export of the adios2 module version and not the DEFAULT. This dynamic linking of the coupler at runtime disrupts XGC and/or GENE's dynamic linking. Thereby leading to errors such as 

GENE
```
 69: /global/homes/d/damilare/wdmapp/wdmapp_coupling/gene_adios2/bin/gene_cori: symbol lookup error: /global/homes/d/damilare/wdmapp/wdmapp_coupling/gene_adios2/bin/gene_cori: undefined symbol: adios2_adios_init_mod_mp_adios2_init_config_

```

XGC
```
/global/homes/d/damilare/wdmapp/wdmapp_coupling/XGC_adios2/xgc-es: symbol lookup error: /global/homes/d/damilare/wdmapp/wdmapp_coupling/XGC_adios2/xgc-es: undefined symbol: adios2_adios_init_mod_mp_adios2_init_config_mpi_
```

In summary, I do not have a working stack due to dynamic linking errors
